tests_e2e/tests/ext_sequencing/ext_sequencing.py

```diff
@@ -0,0 +1,275 @@
+#!/usr/bin/env python3
+
+# Microsoft Azure Linux Agent
+#
+# Copyright 2018 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+#
+# This test adds extensions with multiple dependencies to a VMSS using the 'provisionAfterExtensions' property and
+# validates they are enabled in order of dependencies.
+#
+import copy
+import uuid
+from datetime import datetime
+from typing import List, Dict, Any
+
+from assertpy import fail
+from azure.mgmt.compute.models import VirtualMachineScaleSetVMExtensionsSummary
+
+from tests_e2e.tests.ext_sequencing.ext_seq_test_cases import add_one_dependent_ext_without_settings, add_two_extensions_with_dependencies, \
+    remove_one_dependent_extension, remove_all_dependencies, add_one_dependent_extension, \
+    add_single_dependencies, remove_all_dependent_extensions, add_failing_dependent_extension_with_one_dependency, add_failing_dependent_extension_with_two_dependencies
+from tests_e2e.tests.lib.agent_test import AgentVmssTest, TestSkipped
+from tests_e2e.tests.lib.vm_extension_identifier import VmExtensionIds
+from tests_e2e.tests.lib.logging import log
+from tests_e2e.tests.lib.resource_group_client import ResourceGroupClient
+from tests_e2e.tests.lib.ssh_client import SshClient
+
+
+class ExtSequencing(AgentVmssTest):
+    # Cases to test different dependency scenarios
+    test_cases = [
```

Comment:
other than the run() method, all these properties/methods should be protected (prefixed by '_'), or private ('__')

i tend to use protected even for private members because __ mangles the name and can make debugging harder